tools/syz-declextract/syz-declextract.cpp

```diff
@@ -64,6 +106,230 @@ bool beginsWith(const std::string_view &str, const std::string_view begin) {
 
 bool contains(const std::string_view &str, const std::string_view sub) { return str.find(sub) != std::string::npos; }
 
+bool isPadding(const std::string &fieldName) {
+  return contains(fieldName, "pad") || contains(fieldName, "unused") || contains(fieldName, "_reserved");
+}
+
+const std::string makeArray(const std::string &type, const size_t len = 0) {
+  if (len == 1) {
+    return type;
+  }
+  if (len) {
+    return "array[" + type + ", " + std::to_string(len) + "]";
+  }
+  return "array[" + type + "]";
+}
+
+const std::string makeArray(const std::string &type, const size_t min, const size_t max) {
+  if (min == max) {
+    return makeArray(type, min);
+  }
+  return "array[" + type + ", " + std::to_string(min) + ":" + std::to_string(max) + "]";
+}
+
+const std::string makePtr(const std::string &dir, const std::string &type, bool isOpt = false) {
+  if (type == "void") {
+    return makePtr(dir, makeArray("int8"));
+  }
+  std::string ptr = "ptr[" + dir + ", " + type;
+  if (isOpt) {
+    return ptr + ", opt]";
+  }
+  return ptr + "]";
+}
+
+const std::string makeConst(const std::string &type, const std::string &val = "0") {
+  if (type.empty()) {
+    return "const[" + val + "]";
+  }
+  if (type == "void") {
```

Comment:
This looks fishy. In what cases is it activated?
I can image a pointer to void, but we handle it in makePtr. We don't know it's intptr, right, and just use it as a fallback since we need some type.
It would be clearer to make whoever extracts "void" in the first place to use "intptr", at least it will be clear where it originates.